### ISSUE TITLE
How to Check if Job Exists

### ISSUE THREAD

**Issue by bobdercole** [taskforcesh/bullmq#130]
Hi,

I am currently trying to upgrade from https://github.com/OptimalBits/bull/ and I ran into a bit of a snag. My project relies on this logic:

```typescript
const existingJob = await myQueue.getJob(jobId);

if (existingJob === null) {
	// do some stuff
}
```

The BullMQ type definition for `getJob` looks like this:

```typescript
export declare class QueueGetters extends QueueBase {
    getJob(jobId: string): Promise<Job<any, any>>;
}
```

According to that, `getJob` will always return a job, which makes it hard to check if a job exists.

I noticed `QueueGetters.getJob()` calls `Job.fromId()` internally, which appears to return null if the job is empty.

https://github.com/taskforcesh/bullmq/blob/36726bfb01430af8ec606f36423fc187e4a06fb4/src/classes/job.ts#L143

However, even `job.d.ts` has the same definition:

```typescript
export declare class Job<T = any, R = any> {
    static fromId(queue: QueueBase, jobId: string): Promise<Job<any, any>>;
}
```

Interestingly the v3 compatibility class has exactly what I need.

https://github.com/taskforcesh/bullmq/blob/36726bfb01430af8ec606f36423fc187e4a06fb4/src/classes/compat.ts#L201

Was this functionality purposely changed? Or is it a mistake? If the former, how should I go about getting a job that might not exist?

Thanks!

**Comment by manast** [comment]
probably a mistake in the typings, but it should work the same as v3.

**Comment by bobdercole** [comment]
I created a PR with a fix.

I also noticed some weirdness related to the typings here: https://github.com/bobdercole/bullmq/blob/65183fcf542d0227ec1d4d6637b46b5381331787/src/classes/job.ts#L138

My PR adds `Promise<Job | null>` to the return signature. However, if `jobId` is falsy the method will return undefined, which is not covered by the types I added. Additionally the comment says that `jobId` could be `undefined`, but parameter `jobId: string` states otherwise. Should it be `jobId?: string`?
